src/ed25519/signer.rs

```diff
@@ -13,6 +12,15 @@ pub trait FromSeed: Sized {
     fn from_seed(seed: &[u8]) -> Result<Self, Error>;
 }
 
+/// Trait for Ed25519 signer to generate keys
+pub trait KeyGen {
```

Comment:
Not sure this actually needs a trait, since really it's just generating a uniformly random 32-byte seed, and that should be agnostic to the particular backend (i.e. the implementation is the same regardless of if you're using dalek, ring, or sodiumoxide)